### PR TITLE
[BREAKING] serveResources middleware: Expect *.properties files in UTF-8 by default

### PR DESCRIPTION
**Pull request by RandomByte**
Also fix some tests that expected different unicode escape sequences
for the same input. The used input encoding did not match the specified
propertiesFileSourceEncoding.

BREAKING CHANGE:
If the project a "*.properties" resource originates from cannot be
determined, or if the project does not define a
propertiesFileSourceEncoding configuration or uses a legacy specVersion
(<2.0), the serveResources middleware assumes that the resource is UTF-8
encoded instead of ISO-8859-1.

### CODE REVIEW

File: lib/middleware/serveResources.js [hunk 1]
```diff
@@ -41,9 +41,19 @@ function createMiddleware({resources, middlewareUtil}) {
 				// Special handling for *.properties files escape non ascii characters.
 				const nonAsciiEscaper = require("@ui5/builder").processors.nonAsciiEscaper;
 				const project = resource._project; // _project might not be defined
-				const propertiesFileSourceEncoding = project && project.resources &&
+				let propertiesFileSourceEncoding = project && project.resources &&
 					project.resources.configuration && project.resources.configuration.propertiesFileSourceEncoding;
-				const encoding = nonAsciiEscaper.getEncodingFromAlias(propertiesFileSourceEncoding || "ISO-8859-1");
+
+				if (!propertiesFileSourceEncoding) {
```
[the remark this answers is not in the document]
Comment (RandomByte) [reply lib/middleware/serveResources.js]: As discussed, using semver for these checks is a good idea and should be done in the future.

`project` can be undefined in cases where the middleware (or UI5 Server) is used in a different context than the UI5 CLI. We should always do some defaulting and imho it should be consistent with the latest specVersion.
